lib/buffer.js

```diff
@@ -462,8 +462,11 @@ Buffer.prototype.toString = function() {
   } else {
     result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
+  if (result === undefined) {
+    if (this.length > binding.kStringMaxLength)
+      throw new Error('Size exceeds max string length');
```

Comment:
Yep, that's exact, but I thought that changing the Error type was like a (small but heh) breaking change ; more than just a message anyway.

Should I change it ?